app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt

```diff
@@ -232,6 +234,7 @@ interface CommentsViewHolderViewModel {
 
             comment
                 .compose(takeWhen(this.onViewCommentRepliesButtonClicked))
+                .map { postedSuccessfully?.value ?: it }
```

Comment:
the value for comment and postedSuccessFully should be the same. Why do we need to map here those two values?

Reply:
From my understanding, the comment that was emitted from the comment thread still had the default parentId of -1, which is why the reply was being treated as a root comment. Updating to use the comment item emitted from the postedSuccessfully stream gives us the updated parentId.

Reply:
I would suggest using `Observable.merge` on the declaration for comment observable, something like: 
<img width="929" alt="Screen Shot 2021-07-15 at 4 09 00 PM" src="https://user-images.githubusercontent.com/4083656/125868902-5feed7e7-eb77-4484-b9c6-292ab3f6e463.png">
 in that way you don't need to update several RXChains.

I'll approve though, my comment up there is a suggested improvement 👍

Reply:
Sounds good, thanks for catching this. Just updated 👍🏽